http/operation.go

```diff
@@ -0,0 +1,50 @@
+package http
+
+import (
+	"io"
+	"net/http"
+	"regexp"
+)
+
+type addOperation struct {
+	inner http.RoundTripper
+}
+
+const operationRegex = `((?:query\s)|(?:mutation\s))(\w+)`
+
+func SetGraphqlOperation(inner http.RoundTripper) http.RoundTripper {
+	return &addOperation{
+		inner: inner,
+	}
+}
+
+func (ug *addOperation) RoundTrip(r *http.Request) (*http.Response, error) {
+	values := r.URL.Query()
+	operation := getOperationName(r)
+
+	if operation != "" {
+		values.Add("operation", operation)
+		r.URL.RawQuery = values.Encode()
+	}
+
+	return ug.inner.RoundTrip(r)
+}
+
+func getOperationName(r *http.Request) string {
+	regex := regexp.MustCompile(operationRegex)
+	getBody := r.GetBody
+
+	copyBody, err := getBody()
```

Comment:
I'm not sure, but shouldn't be `copyBody` or `b` closed after reading them? Even if they are actually copies of the request body 🤔